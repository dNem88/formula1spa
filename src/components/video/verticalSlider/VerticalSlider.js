import React from 'react'
import styles from './verticalSlider.module.css'
import VideoInfo from '../../common/videoInfo/VideoInfo'
import Video from '../../common/video/Video'

function VerticalSlider({videos}) {


    return (
        <div className={styles['slider']}>
            <h3 className={styles['more-videos-title']}>More videos</h3>
            <div className={styles['relative-wrapper']}>
                <div className={styles['slider-container-absolute']}>
                    {videos.slice(0,4).map(x => {
                        return <div key={x._id} className={styles['video-wrapper']}>
                                    <Video url={x.url}/>
                                    <VideoInfo  _id={x._id} title={x.title}/>
                                </div>
                                    })}
                </div>
            </div> 
        </div>
    )
}

export default VerticalSlider