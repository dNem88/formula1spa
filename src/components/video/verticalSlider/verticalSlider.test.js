import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import VerticalSlider from './VerticalSlider'



jest.mock('../../common/video/Video', () => () => 'Video')
jest.mock('../../common/videoinfo/VideoInfo', () => () => 'VideoInfo')
const mockProps = {videos: [
    {url: 'test-url', _id: 'test-id', }
]}

describe('Testing VericalSlider', () => {
    
    it('Render Properly', () => {
        const snapshot = renderer.create(
        <MemoryRouter>
            <VerticalSlider {...mockProps}/>
        </MemoryRouter>
    ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('div')
        expect(snapshot.props.className).toBe('slider')
        expect(snapshot.children.length).toBe(2)
        expect(snapshot.children[0].children[0]).toBe('More videos')
        expect(snapshot.children[1].type).toBe('div')
        expect(snapshot.children[1].props.className).toBe('relative-wrapper')
        let videosContainer = snapshot.children[1].children[0]
        expect(videosContainer.type).toBe('div')
        expect(videosContainer.props.className).toBe('slider-container-absolute')
    })
})