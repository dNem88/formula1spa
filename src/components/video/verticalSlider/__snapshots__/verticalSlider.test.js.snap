// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing VericalSlider Render Properly 1`] = `
<div
  className="slider"
>
  <h3
    className="more-videos-title"
  >
    More videos
  </h3>
  <div
    className="relative-wrapper"
  >
    <div
      className="slider-container-absolute"
    >
      <div
        className="video-wrapper"
      >
        Video
        VideoInfo
      </div>
    </div>
  </div>
</div>
`;
